File: content-sharing-backend/csapp/src/main/java/com/app/csapp/dtos/ReactDTO.java
package com.app.csapp.dtos;


import com.app.csapp.enums.ReactEnums;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ReactDTO {

    @JsonProperty("user_id")
    @NotNull(message =  "User id cannot be null")
    private Long userId;

    @JsonProperty("picture_id")
    @NotNull(message = "Picture id cannot be null")
    private Long pictureId;

    @NotBlank
    @JsonProperty("type_react")
    private ReactEnums typeReact;

    @JsonProperty("content")
    private String content;
}

File: frontend/pinterest-angular/src/app/dtos/reacts/react.dto.ts
import {
    IsNotEmpty
} from 'class-validator';

export class ReactDTO {
    @IsNotEmpty()
    user_id: number;

    @IsNotEmpty()
    picture_id: number; 

    type_react: number;

    @IsNotEmpty()
    content: string;

    constructor(data: any) {
        this.user_id = data.user_id;
        this.picture_id = data.picture_id;
        this.type_react = 0;
        this.content = data.content;
    }
}